packages/itwin/measure-tools/src/api/Measurement.ts

```diff
@@ -285,6 +304,66 @@ export abstract class Measurement {
     this.onTransientIdChanged(prevId);
   }
 
+  public get drawingMetaData(): DrawingMetaData | undefined {
+    return this._drawingMetaData;
+  }
+
+  public set drawingMetaData(data: DrawingMetaData | undefined) {
+    this._drawingMetaData = data;
+  }
+
+  public set sheetViewId(id: string | undefined) {
+    this._sheetViewId = id;
+  }
+
+  public get sheetViewId(): string | undefined {
+    return this._sheetViewId;
+  }
+
+  public set drawingId(id: string | undefined) {
+    if (this._drawingMetaData)
+      this._drawingMetaData.drawingId = id;
+    else
+      this._drawingMetaData = { drawingId: id };
+  }
+
+  public get drawingId(): string | undefined {
+    return this._drawingMetaData?.drawingId;
+  }
+
+  public set drawingOrigin(origin: Point2d | undefined) {
+    if (this._drawingMetaData)
+      this._drawingMetaData.origin = origin;
+    else
+      this._drawingMetaData = { origin };
+  }
+
+  public get drawingOrigin(): Point2d | undefined {
+    return this._drawingMetaData?.origin;
+  }
+
+  public set drawingExtents(extents: Point2d | undefined) {
+    if (this._drawingMetaData)
+      this._drawingMetaData.extents = extents;
+    else
+      this._drawingMetaData = { extents };
+  }
+
+  public get drawingExtents(): Point2d | undefined {
+    return this._drawingMetaData?.extents;
+  }
+
+  public setWorldScale(scale: number | undefined) {
```

Comment:
Having `setWorldScale` as a function while the others are getter/setter is off. Note a lot of the base properties have an `onXYZChanged` that subclasses can override, I would do the same for the drawing metadata properties (or maybe a single on changed?)

Also, I still think it's reasonable that at least worldScale/origin should be required properties on the metadata. Probably do some validation for the world scale, since usually scales cannot be zero (in this case, probably also not negative).